docs/concepts-advanced/environment-variables.md

```diff
@@ -1,170 +1,352 @@
-      typeId:463,
-      scope:RUNTIME,
-      name:"MYVARIABLENAME",
-      value:"MyVariableValue"
+    ```bash title=""
+    lagoon add variable --organization-name lagoon-demo-org --scope runtime \
+    --name SSMTP_MAILHUB --value mailhub.example.com
+    ```
+
+=== "Lagoon API"
+
+    ```graphql title=""
+    mutation {
+      addOrUpdateEnvVariableByName(
+        input: {
+          organization: "lagoon-demo-org"
+          scope: RUNTIME
+          name: "SSMTP_MAILHUB"
+          value: "mailhub.example.com"
+        }
+      ) {
+        id
+        name
+        scope
+        value
+      }
     }
-  ) {
-    id
-  }
-}
-```
+    ```
+
+#### Project
+
+Project env vars are set for all environments in a project. They can be overridden by environment env vars.
+
+An example of setting a `project` env var:
+
+=== "Lagoon CLI"
+
+    ```bash title=""
+    lagoon add variable --project lagoon-demo --scope runtime \
+    --name PHP_MEMORY_LIMIT --value 400M
+    ```
+
+=== "Lagoon API"
+
+    ```graphql title=""
+    mutation {
+      addOrUpdateEnvVariableByName(
+        input: {
+          project: "lagoon-demo"
+          scope: RUNTIME
+          name: "PHP_MEMORY_LIMIT"
+          value: "400M"
+        }
+      ) {
+        id
+        name
+        scope
+        value
+      }
+    }
+    ```
+
+#### Environment
+
+Environment env vars are set for a single Lagoon environment.
+
+An example of setting an `environment` env var:
+
+=== "Lagoon CLI"
+
+    ```bash title=""
+    lagoon add variable --project lagoon-demo --environment dev \
+    --scope runtime --name LOG_LEVEL --value DEBUG
+    ```
+
+=== "Lagoon API"
+
+    ```graphql title=""
+    mutation {
+      addOrUpdateEnvVariableByName(
+        input: {
+          project: "lagoon-demo"
+          environment: "dev
+          scope: RUNTIME
+          name: "LOG_LEVEL"
+          value: "DEBUG"
+        }
+      ) {
+        id
+        name
+        scope
+        value
+      }
+    }
+    ```
+
+### Scopes
+
+Scopes are used to set an env var for a specific context. An env var can have a scope of `buildtime`, `runtime`, `global`, `container_registry` or `internal_container_registry`.
+
+#### Buildtime
 
-This defines a environment ID `546` specific runtime variable (available only in that specific environment):
-
-```graphql title="Define environment ID"
-mutation addRuntimeEnv {
-  addEnvVariable(
-    input:{
-      type:ENVIRONMENT,
-      typeId:546,
-      scope:RUNTIME,
-      name:"MYVARIABLENAME",
-      value:"MyVariableValue"
+Buildtime env vars are only available during Docker image builds, and can be accessed by using the `ARG` instruction. Check the [Docker documention on `ARG`](https://docs.docker.com/reference/dockerfile/#arg) for all the ways to use this method.
+
+Typically the `ARG` will go after the FROM. Read [the Docker documentation about ARG and FROM](https://docs.docker.com/engine/reference/builder/#understand-how-arg-and-from-interact).
```

Comment:
I wonder if it wouldn't be useful for us to have some kind of example illustrating the difference between build and runtime variables?

Reply:
I'm open to suggestions. I picked the names of env vars that would make sense for the scopes used, but that assumes the reader would recognize them.